pkg/controller/machine/machine_controller_phases.go

```diff
@@ -122,8 +130,21 @@ func (r *ReconcileMachine) reconcileExternal(ctx context.Context, m *v1alpha2.Ma
 		obj.SetOwnerReferences(util.EnsureOwnerRef(obj.GetOwnerReferences(), machineOwnerRef))
 		if err := r.Patch(ctx, obj, objPatch); err != nil {
 			return nil, errors.Wrapf(err,
-				"failed to set OwnerReference on %s %q for Machine %q in namespace %q",
-				path.Join(ref.APIVersion, ref.Kind), ref.Name, m.Name, m.Namespace)
+				"failed to set OwnerReference on %v %q for Machine %q in namespace %q",
+				obj.GroupVersionKind(), ref.Name, m.Name, m.Namespace)
+		}
+	}
+
+	// Add watcher for external object, if there isn't one already.
+	_, loaded := r.externalWatchers.LoadOrStore(obj.GroupVersionKind().String(), struct{}{})
+	if !loaded && r.controller != nil {
+		err := r.controller.Watch(
```

Comment:
Could you add a log message saying we're attempting to watch obj.GVK?

Reply:
Added!